lib/AnyEvent/HTTP/Server/Req.pm

```diff
@@ -319,6 +319,20 @@ use Digest::SHA1 'sha1';
 				$h->{'content-type'} = 'text/html; charset=utf-8';
 			}
 			my $nh = delete $h->{NotHandled};
+
+			# set multiple cookies
+			my $cookie_key;
+			grep { lc $_ eq 'set-cookie' and $cookie_key = $_; } keys %$h;
+
+			if(defined $h->{$cookie_key} and ref $h->{$cookie_key} eq 'ARRAY') {
+				my $cookies = delete $h->{$cookie_key};
+				my $set_cookie = '';
+
+				#set-cookie is not in @hdr
+				$set_cookie .= "\u\Lset-cookie\E: ".$_.$LF for(@{$cookies});
```

Comment:
1. Canonical form is `Set-Cookie`. No need to use `\u\L`
2. Just push every line of set-cookie into `@bad`, concatenation is useless here